Celeste.Mod.mm/Patches/Monocle/Tracker.cs

```diff
@@ -137,14 +137,17 @@ public static void AddTypeToTracker(Type type, Type trackedAs = null, params Typ
         }
 
         private static bool AddSpecificType(Type type, Type trackedAsType, Dictionary<Type, List<Type>> tracked, HashSet<Type> knownTypes) {
+            // Make sure the tracker knows about both the type and trackedAs type, to fix scenarios like `[TrackedAs(typeof(UntrackedType))]`
+            bool updated = false;
+            if (knownTypes is not null) {
+                updated = knownTypes.Add(type);
+                updated |= knownTypes.Add(trackedAsType);
+            }
+            // check IsAbstract later, make AddTypeToTracker(type: typeof(Celeste.CutsceneEntity), trackedAs: null, inheritAll: true) work properly
             if (type.IsAbstract) {
-                return false;
+                return updated;
             }
```

Comment:
```suggestion
            // Make sure the tracker knows about trackedAs type, to fix scenarios like `[TrackedAs(typeof(UntrackedType))]
            // Vanilla adds the target type (`trackedAsType`) regardless of whether it's abstract or not
            bool updated = knownTypes.Add(trackedAsType);
            if (type.IsAbstract) {
                return updated;
            }
```
Why not just get rid of the `knownTypes.Add(type)`? As far as my understanding goes when a type `T` is in `knownTypes` it means that the tracker is aware of it (not just that it is on the keys of `tracked`) and `T` is tracked, `tracker.GetEntities<T>()` will return instances of `T`. (In other words: `tracked[typeof(T)].Contains(typeof(T))` is true for `T` when `T` is in `knownTypes`.)

Furthermore, `trackedAsType` should always be added to `knownTypes` since it will be present on some list in the values (not keys) of `tracked`.

Reply:
I agree. I reviewed the commit history again, and it looks like we were very close to a correct solution earlier